### PR TITLE
docs: split API.md in different files

### PR DESCRIPTION
**Pull request by matallo**
as done in cartodb.js, I've splitted `API.md` in different files for easier docs exploration

feel free to comment or rearrange as needed

CR @rochoa @csobier

### CODE REVIEW

File: doc/API.md [hunk 1]
```diff
@@ -10,326 +10,12 @@ There are two main situations in which you would want to use the SQL API:
...
-
-You can also just do `select user` using the SQL API (without an API key to get the publicuser name and with an API key to get the CartoDB user's PostgreSQL user name), to determine the name of the corresponding PostgreSQL user.
-
-### Could I configure my CartoDB database permissions exactly the same way I could on my own PostgreSQL instance?
-
-Yes, through using GRANT statements to the SQL API. There are a few caveats to be aware of, including the aforementioned naming differences. Also, you'll be limited to permissions a user has with their own tables. Users don’t have PostgreSQL superuser privileges. So they can’t be creating languages, or C functions, or anything that requires superuser or CREATEUSER privileges.
+* [Authentication](/CartoDB/CartoDB-SQL-API/blob/master/doc/authentication.md)
```
Comment (rochoa) [inline doc/API.md]: Can we use relative links for this?

If I'm working on a branch I want to follow links in that branch and no absolute links pointing to master documentation.

**Comment by matallo** [comment]
roger, I also did this change in the rest of the repos

**Comment by matallo** [comment]
bump, still pending :+1: @rochoa @csobier

**Comment by csobier** [comment]
:+1:

**Comment by matallo** [comment]
:sparkles:

**Pull request merged**